### PR TITLE
Upgrade Probot

### PR DESCRIPTION
**Pull request by jameesjohn**
This PR upgrades oppiabot-stale to the latest on probot-stale and uses the latest version of probot to resolve compatibility issues.

### CODE REVIEW

**Comment by jameesjohn** [comment]
PTAL @ankita240796 @kevinlee12. Thanks!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
```
Comment (kevinlee12) [inline .github/workflows/ci.yml]: Why are there so many node versions?

**Review by kevinlee12 — COMMENTED** [review]
I have just a comment, I'll defer to @ankita240796 for the i ndepth review.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jameesjohn) [reply .github/workflows/ci.yml]: I think this is just done to ensure that the build supports all the different node versions.
It is part of the update from https://github.com/probot/stale

**Comment by kevinlee12** [comment]
got it. we aren’t running it on different versions, so it’s not necessary. I want us to get into the habit of running CI only when it’s necessary.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ankita240796) [reply .github/workflows/ci.yml]: Why not just the latest?

File: test/stale.test.js [hunk 2]
```diff
@@ -185,14 +185,144 @@ describe('stale', () => {
...
       await stale.markAndSweep('pulls')
       expect(stale.getClosable).not.toHaveBeenCalled()
     }
   )
+
+  describe('mark', () => {
+    test(
+      'should not mark issue if it is already closed',
+      async () => {
+        let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Line too long, break.

File: test/stale.test.js [hunk 3]
```diff
@@ -185,14 +185,144 @@ describe('stale', () => {
...
+
+        await stale.mark('issues')
+        expect(stale.markIssue).not.toHaveBeenCalled()
+      }
+    )
+
+    test(
+      'should not mark issue if it is locked',
+      async () => {
+        let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Ditto.

File: test/stale.test.js [hunk 4]
```diff
@@ -185,14 +185,144 @@ describe('stale', () => {
...
+
+        await stale.mark('issues')
+        expect(stale.markIssue).not.toHaveBeenCalled()
+      }
+    )
+
+    test(
+      'should mark issue if it is open',
+      async () => {
+        let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Ditto.

File: test/stale.test.js [hunk 5]
```diff
@@ -185,14 +185,144 @@ describe('stale', () => {
...
+      }
+    )
+  })
+
+  describe('sweep', () => {
+    test(
+      'should not close issue if it is already closed',
+      async () => {
+        const staleLabel = 'stale'
+        let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Ditto.

File: test/stale.test.js [hunk 6]
```diff
@@ -185,14 +185,144 @@ describe('stale', () => {
...
+        await stale.sweep('issues')
+        expect(stale.close).not.toHaveBeenCalled()
+      }
+    )
+
+    test(
+      'should not close issue if it is locked',
+      async () => {
+        const staleLabel = 'stale'
+        let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Ditto.

File: test/stale.test.js [hunk 7]
```diff
@@ -185,14 +185,144 @@ describe('stale', () => {
...
+        await stale.sweep('issues')
+        expect(stale.close).not.toHaveBeenCalled()
+      }
+    )
+
+    test(
+      'should close issue if it is open',
+      async () => {
+        const staleLabel = 'stale'
+        let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Ditto.

File: test/stale.test.js [hunk 8]
```diff
@@ -38,17 +38,17 @@ describe('stale', () => {
...
     // Mock out GitHub client
-    robot.auth = () => Promise.resolve(github)
+    app.auth = () => Promise.resolve(github)
   })
 
   test(
     'removes the stale label and ignores if it has already been removed',
     async () => {
-      let stale = new Stale(github, {perform: true, owner: 'probot', repo: 'stale', logger: robot.log})
+      let stale = new Stale(github, { perform: true, owner: 'probot', repo: 'stale', logger: app.log })
```
Comment (ankita240796) [inline test/stale.test.js]: Ditto here and below.

File: lib/schema.js [hunk 9]
```diff
@@ -8,6 +8,9 @@ const fields = {
     .error(() => '"daysUntilClose" must be a number or false')
     .description('Number of days of inactivity before a stale Issue or Pull Request is closed. If disabled, issues still need to be closed manually, but will remain marked as stale.'),
 
+  onlyLabels: Joi.alternatives().try(Joi.any().valid(null), Joi.array().single())
```
Comment (ankita240796) [inline lib/schema.js]: Ditto. Please fix everywhere.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ankita240796) [reply .github/workflows/ci.yml]: Actually, I am fine with not having ci tests on this for repo for now since we are anyways planning to deprecate it.

**Review by ankita240796 — CHANGES_REQUESTED** [review]
Hi @jameesjohn, looks good. Few minor nits, PTAL. Thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jameesjohn) [reply .github/workflows/ci.yml]: Okay, I'd be removing this

File: lib/schema.js (hunk 9 above)
Comment (jameesjohn) [reply lib/schema.js]: Done.

File: test/stale.test.js (hunk 8 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

File: test/stale.test.js (hunk 2 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

File: test/stale.test.js (hunk 3 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

File: test/stale.test.js (hunk 4 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

File: test/stale.test.js (hunk 5 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

File: test/stale.test.js (hunk 6 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

File: test/stale.test.js (hunk 7 above)
Comment (jameesjohn) [reply test/stale.test.js]: Done.

**Comment by jameesjohn** [comment]
PTAL @ankita240796

**Review by ankita240796 — APPROVED** [review]
LGTM, Thanks @jameesjohn! Just to confirm, have you tested this?

**Comment by jameesjohn** [comment]
> Just to confirm, have you tested this?

I have not been able to test this because this is a dependency in oppiabot and can only be tested after this PR has been merged then npm update will be get the latest version of this.

**Comment by ankita240796** [comment]
> > Just to confirm, have you tested this?
> 
> I have not been able to test this because this is a dependency in oppiabot and can only be tested after this PR has been merged then npm update will be get the latest version of this.

Makes sense. Will merge and deploy. Thanks!

**Pull request merged**